pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -74,8 +77,9 @@ func updateSecurityRuleNameAndPriority(existingRules []armnetwork.SecurityRule,
 		if rule.Properties == nil {
 			continue
 		}
+		// skip priority update for default 4096 deny all rule. newRules consists of only Nephe rules.
 		if *rule.Properties.Priority == vnetToVnetDenyRulePriority {
-			defaultRulesByName[*rule.Name] = rule
+			defaultAndUserRules = append(defaultAndUserRules, rule)
```

Comment:
Default VNET deny rule will be added twice in the slice defaultAndUserRules. 
Does the azure client handle it?